### PR TITLE
go: Add docs for module-aware build mode.

### PR DESCRIPTION
**Pull request by dennwc**
Current build method described in `go-get` section relies on legacy `GOPATH` mode.

Also, there's a `flatpak-go-vendor-generator.py` that generates `vendor` directory, but refers to https://github.com/golang/go/issues/35922 for a better way to extract direct URLs.

This change adds documentation for a fully modules-aware method (with examples) which supports all edge cases (like `replace` directives in `go.mod`) and downloads ZIP archives of the dependencies from the Go modules proxy.

It requires Go tooling on the host (specifically `go mod download -json`) and thus assumes that installing an external Go tools for generation is acceptable.

### CODE REVIEW

File: go-modules/README.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+This method generates sources section with references to ZIP files hosted by the Go modules proxy.
+
+## Usage
+
+Example manifest (yaml):
+
+```yaml
+app-id: io.github.golang.tools.gorename
+runtime: org.freedesktop.Platform
+runtime-version: '22.08'
```
Comment (hfiguiere) [inline go-modules/README.md:13]:
```suggestion
runtime-version: '23.08'
```

File: go-modules/README.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+sdk: org.freedesktop.Sdk
+sdk-extensions:
+  - org.freedesktop.Sdk.Extension.golang
+command: echo "Done"
```
Comment (hfiguiere) [inline go-modules/README.md:17]: this is not valid. it expect the name of an executable.

File: go-modules/README.md [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+3. Append generated `go.mod.yaml` file to `sources` section of the manifest.
+4. Add `modules.txt` to your Flatpak project.
+5. Build/install your binary in vendor mode (with `-mod=vendor`).
+
+## Final manifest example
+
+```yaml
+app-id: io.github.golang.tools.gorename
+runtime: org.freedesktop.Platform
+runtime-version: '22.08'
```
Comment (hfiguiere) [inline go-modules/README.md:42]:
```suggestion
runtime-version: '23.08'
```

File: go-modules/README.md [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+sdk: org.freedesktop.Sdk
+sdk-extensions:
+  - org.freedesktop.Sdk.Extension.golang
+command: echo "Done"
```
Comment (hfiguiere) [inline go-modules/README.md:46]: same as above.

Comment (hfiguiere) [reply go-modules/README.md:46]:
```suggestion
command: gorename
```

I assume it's the name and it is installed.

File: go-modules/README.md (hunk 2 above)
Comment (hfiguiere) [reply go-modules/README.md:17]:
```suggestion
command: gorename
```

**Comment by dennwc** [comment]
@hfiguiere Done! Rebased and applied the changes you mentioned.

**Pull request merged**